### PR TITLE
High velocity filter

### PR DESCRIPTION
**Pull request by alhighfidelity**
Pull request for testing only.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1607/HighFidelity-Beta-PR10953-cb81158073d88e492119f93dc67c5c7001be2db2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1607/HighFidelity-Beta-PR10953-cb81158073d88e492119f93dc67c5c7001be2db2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1626/HighFidelity-Beta-PR10953-f83942b2b3948b7863590a8bfb25d5464b610c2c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1626/HighFidelity-Beta-PR10953-f83942b2b3948b7863590a8bfb25d5464b610c2c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1636/HighFidelity-Beta-PR10953-a0dbf1c6533a967405485da09a10649696695455.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1641/HighFidelity-Beta-PR10953-fa883658129761ab8ded0635a61b2ade45de28da.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1641/HighFidelity-Beta-PR10953-fa883658129761ab8ded0635a61b2ade45de28da.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1643/HighFidelity-Beta-PR10953-f5a19c4fd90ed50469578195b09ff5f9e1c9651e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1643/HighFidelity-Beta-PR10953-f5a19c4fd90ed50469578195b09ff5f9e1c9651e.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1707/HighFidelity-Beta-PR10953-a936888f32ac6af809cb13146dfbffc4f26f8d4b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1707/HighFidelity-Beta-PR10953-a936888f32ac6af809cb13146dfbffc4f26f8d4b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1777/HighFidelity-Beta-PR10953-5582cd4342dbafff5a6a1337bc452427162b652c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1777/HighFidelity-Beta-PR10953-5582cd4342dbafff5a6a1337bc452427162b652c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1782/HighFidelity-Beta-PR10953-46595efa2cf7e81ee584e60d022fb50e8088f63d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1782/HighFidelity-Beta-PR10953-46595efa2cf7e81ee584e60d022fb50e8088f63d.exe

File: libraries/controllers/src/controllers/impl/filters/HighVelocityFilter.cpp [hunk 1]
```diff
@@ -0,0 +1,660 @@
...
+        ret.rotation = rot;
+
+        return ret;
+    }
+
+
+
+
+
+    glm::vec3 HighVelocityFilter::diff(glm::vec3 v1, glm::vec3 v2) const {
```
Comment (ZappoMan) [inline libraries/controllers/src/controllers/impl/filters/HighVelocityFilter.cpp]: why make this a function... just use the vec3's directly where you need them

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1833/HighFidelity-Beta-PR10953-e4da20109d1b1be9fe0a8bc0500c2e82bc256f80.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1833/HighFidelity-Beta-PR10953-e4da20109d1b1be9fe0a8bc0500c2e82bc256f80.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1970/HighFidelity-Beta-PR10953-51efd3b71dc3a3e66800d1ba2d7010f1b4ecca1f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1970/HighFidelity-Beta-PR10953-51efd3b71dc3a3e66800d1ba2d7010f1b4ecca1f.dmg

**Comment by sethalves** [comment]
Here's some of the build output related to the build failure:
`00:21:37 [ 32%] Building CXX object libraries/networking/CMakeFiles/networking.dir/src/NetworkPeer.cpp.o
00:21:38 /home/jenkins/workspace/pr-build/label/ubuntu/libraries/controllers/src/controllers/impl/filters/HighVelocityFilter.cpp: In member function ‘virtual controller::Pose controller::HighVelocityFilter::apply(controller::Pose) const’:
00:21:38 /home/jenkins/workspace/pr-build/label/ubuntu/libraries/controllers/src/controllers/impl/filters/HighVelocityFilter.cpp:216:39: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
00:21:38                      for (int i = 0; i < len; i++) {
00:21:38                                        ^
00:21:38 /home/jenkins/workspace/pr-build/label/ubuntu/libraries/controllers/src/controllers/impl/filters/HighVelocityFilter.cpp:238:39: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
00:21:38                      for (int i = 0; i < len; i++) {
00:21:38                                        ^
00:21:38 /home/jenkins/workspace/pr-build/label/ubuntu/libraries/controllers/src/controllers/impl/filters/HighVelocityFilter.cpp:276:39: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
00:21:38                      for (int i = 0; i < len; i++) {
00:21:38                                        ^
00:21:38 /home/jenkins/workspace/pr-build/label/ubuntu/libraries/controllers/src/controllers/impl/filters/HighVelocityFilter.cpp:285:39: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
00:21:38                      for (int i = 0; i < len; i++) {
00:21:38                                        ^
00:21:38 /home/jenkins/workspace/pr-build/label/ubuntu/libraries/controllers/src/controllers/impl/filters/HighVelocityFilter.cpp: In member function ‘float controller::HighVelocityFilter::maxIndex(glm::vec3) const’:
00:21:38 /home/jenkins/workspace/pr-build/label/ubuntu/libraries/controllers/src/controllers/impl/filters/HighVelocityFilter.cpp:418:17: error: ‘fabsf’ is not a member of ‘std’`

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1973/HighFidelity-Beta-PR10953-8aec6363f118b431848226611b71340e35fc3873.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1973/HighFidelity-Beta-PR10953-8aec6363f118b431848226611b71340e35fc3873.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1987/HighFidelity-Beta-PR10953-d0fd75c7887c2cd640e09ed8e501887c95e9b5cd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1987/HighFidelity-Beta-PR10953-d0fd75c7887c2cd640e09ed8e501887c95e9b5cd.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2003/HighFidelity-Beta-PR10953-3695cbc07bed924be248e583d0e95e27859f9192.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2013/HighFidelity-Beta-PR10953-c4dcabcd6c88015c00a3754c836d7af82c6526a5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2013/HighFidelity-Beta-PR10953-c4dcabcd6c88015c00a3754c836d7af82c6526a5.exe

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2014/HighFidelity-Beta-PR10953-c4dcabcd6c88015c00a3754c836d7af82c6526a5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2014/HighFidelity-Beta-PR10953-c4dcabcd6c88015c00a3754c836d7af82c6526a5.exe

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2015/HighFidelity-Beta-PR10953-c4dcabcd6c88015c00a3754c836d7af82c6526a5.dmg

**Comment by marko8904** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2064/HighFidelity-Beta-PR10953-9967e928d5e556f52652ae4567d0848df391a609.dmg

**Comment by alhighfidelity** [comment]
"build this please"

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2073/HighFidelity-Beta-PR10953-7f80fdb15d167edfb282c5e7711ab5ec1057ddd4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2074/HighFidelity-Beta-PR10953-66a6ec98c51d9a87ab9c927fb0725f4548d4e4cb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2074/HighFidelity-Beta-PR10953-66a6ec98c51d9a87ab9c927fb0725f4548d4e4cb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2100/HighFidelity-Beta-PR10953-3964dd5200e922539945e7d05ba8cd03e379505b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2100/HighFidelity-Beta-PR10953-3964dd5200e922539945e7d05ba8cd03e379505b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2154/HighFidelity-Beta-PR10953-d65f5449c11469648ae9bef0fd74b4c2c84e161c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2154/HighFidelity-Beta-PR10953-d65f5449c11469648ae9bef0fd74b4c2c84e161c.exe

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2192/HighFidelity-Beta-PR10953-1990070356f2493771da5bfce03817daeb14e71e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2192/HighFidelity-Beta-PR10953-1990070356f2493771da5bfce03817daeb14e71e.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2194/HighFidelity-Beta-PR10953-ac8a6bcaa573120381a475ccfae0a2c3c4b8411a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2194/HighFidelity-Beta-PR10953-ac8a6bcaa573120381a475ccfae0a2c3c4b8411a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2196/HighFidelity-Beta-PR10953-4674ec1279aec64f3e2053af861508770f68f9da.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2196/HighFidelity-Beta-PR10953-4674ec1279aec64f3e2053af861508770f68f9da.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2213/HighFidelity-Beta-PR10953-d1b07dfbf5b2f715ca8d803f2ffa6c884d2bda58.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2213/HighFidelity-Beta-PR10953-d1b07dfbf5b2f715ca8d803f2ffa6c884d2bda58.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2223/HighFidelity-Beta-PR10953-178fdfd7dca67a69d26ed61afface55a8e99c8bd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2223/HighFidelity-Beta-PR10953-178fdfd7dca67a69d26ed61afface55a8e99c8bd.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2233/HighFidelity-Beta-PR10953-1243043b93acdab225da987b770fa7033ee1f9e3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2233/HighFidelity-Beta-PR10953-1243043b93acdab225da987b770fa7033ee1f9e3.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2236/HighFidelity-Beta-PR10953-0b06740702071d81538ebafb5e56ffebf067cc2a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2236/HighFidelity-Beta-PR10953-0b06740702071d81538ebafb5e56ffebf067cc2a.exe

**Comment by alhighfidelity** [comment]
"build this please"

**Comment by anshuman64** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2239/HighFidelity-Beta-PR10953-3b40043a6ee2d2a7e830025bd98d638a56b6e0cb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2239/HighFidelity-Beta-PR10953-3b40043a6ee2d2a7e830025bd98d638a56b6e0cb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2245/HighFidelity-Beta-PR10953-84aa83626d9c040f8ee5b43cab76b1a6a34d7b45.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2245/HighFidelity-Beta-PR10953-84aa83626d9c040f8ee5b43cab76b1a6a34d7b45.exe

**Comment by alhighfidelity** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2266/HighFidelity-Beta-PR10953-43151bd859b826972fc00415102928b53f9a94dc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2266/HighFidelity-Beta-PR10953-43151bd859b826972fc00415102928b53f9a94dc.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2271/HighFidelity-Beta-PR10953-88e6b891e534a182c74e7896cc0c781b47d73ba2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2271/HighFidelity-Beta-PR10953-88e6b891e534a182c74e7896cc0c781b47d73ba2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2278/HighFidelity-Beta-PR10953-64e2406c4567a679fed725a8bce6278316130c53.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2278/HighFidelity-Beta-PR10953-64e2406c4567a679fed725a8bce6278316130c53.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2285/HighFidelity-Beta-PR10953-4765717d0170ac3f372fd56a3e914b579d61308b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2285/HighFidelity-Beta-PR10953-4765717d0170ac3f372fd56a3e914b579d61308b.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2286/HighFidelity-Beta-PR10953-87f03ee7d6053e67b39f9bece7da15672f31bdcb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2286/HighFidelity-Beta-PR10953-87f03ee7d6053e67b39f9bece7da15672f31bdcb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2287/HighFidelity-Beta-PR10953-6b1b0685ba9618f6c52a69545caf4a09599750a0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2287/HighFidelity-Beta-PR10953-6b1b0685ba9618f6c52a69545caf4a09599750a0.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2288/HighFidelity-Beta-PR10953-5d696224eebe5e20e9c320ebf63790fea2d11ea3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2288/HighFidelity-Beta-PR10953-5d696224eebe5e20e9c320ebf63790fea2d11ea3.exe

**Comment by alhighfidelity** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2291/HighFidelity-Beta-PR10953-2bff281f55dedddf4d950f242bf65bccb628a05a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2291/HighFidelity-Beta-PR10953-2bff281f55dedddf4d950f242bf65bccb628a05a.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2322/HighFidelity-Beta-PR10953-3da3460322cb13f6611b5b75e9f6a15497029d2f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2322/HighFidelity-Beta-PR10953-3da3460322cb13f6611b5b75e9f6a15497029d2f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2324/HighFidelity-Beta-PR10953-336cb34862ed15a72b57f7f448b8b245c92c66bb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2324/HighFidelity-Beta-PR10953-336cb34862ed15a72b57f7f448b8b245c92c66bb.exe

**Pull request closed without merging**